pkg/queue/concurrency_state.go

```diff
@@ -85,3 +88,55 @@ func ConcurrencyStateHandler(logger *zap.SugaredLogger, h http.Handler, pause, r
 		<-done
 	}
 }
+
+// Pause sends to an endpoint when request counts drop to zero
+func Pause(endpoint string) error {
+	action := ConcurrencyStateMessageBody{Action: "pause"}
+	bodyText, err := json.Marshal(action)
+	if err != nil {
+		return fmt.Errorf("unable to create request body: %w", err)
+	}
+	body := bytes.NewBuffer(bodyText)
+	req, err := http.NewRequest(http.MethodPost, endpoint, body)
+	if err != nil {
+		return fmt.Errorf("unable to create request: %w", err)
+	}
+	req.Header.Add("Token", "nil") // TODO: use serviceaccountToken from projected volume
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("unable to post request: %w", err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("expected 200 response, got: %d: %s", resp.StatusCode, resp.Status)
+	}
+
+	return nil
+}
+
+// Resume sends to an endpoint when request counts increase from zero
+func Resume(endpoint string) error {
+	action := ConcurrencyStateMessageBody{Action: "resume"}
+	bodyText, err := json.Marshal(action)
+	if err != nil {
+		return fmt.Errorf("unable to create request body: %w", err)
+	}
+	body := bytes.NewBuffer(bodyText)
+	req, err := http.NewRequest(http.MethodPost, endpoint, body)
+	if err != nil {
+		return fmt.Errorf("unable to create request: %w", err)
+	}
+	req.Header.Add("Token", "nil") // TODO: use serviceaccountToken from projected volume
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("unable to post request: %w", err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("expected 200 response, got: %d: %s", resp.StatusCode, resp.Status)
+	}
+
+	return nil
+}
```

Comment:
This is the exact same function as above modulo `Action: "resume"`. Shall we deduplicate?

Reply:
Yes, that makes sense. I recall discussing in one of the other PRs keeping the pause/resume functions separate for testing purposes, but no need for identical functions. I can dedupe to a single function (say `func concurrencyStateRequest(b ConcurrencyStateMessageBody) error {}` (removing endpoint as per comment below).

One question -- would it still be worthwhile to keep distinct pause/resume functions (where each would just call the `concurrencyStateRequest` function with the relevant parameter? I could see it help with readability, but I may be overthinking things a bit too...

Reply:
Yeah it's fine to have both as readability wrappers IMO.

Reply:
Though I could also see the form ultimately being:

In main.go
```golang
resume := concurrencyStateRequest(endpoint, ConcurrencyStateMessageBody{Action: "resume"})
pause := concurrencyStateRequest(endpoint, ConcurrencyStateMessageBody{Action: "pause"})
ConcurrencyStateHandler(logger, handler, resume, pause)
```

Reply:
Ahh, I misread the comment below... yes, passing the endpoint directly to the resume/pause functions (and skipping the Handler function entirely) makes total sense. Thanks for the clarification!